cmd/kubectl_token_test.go

```diff
@@ -0,0 +1,114 @@
+package cmd
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	apiv3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
+	"github.com/stretchr/testify/assert"
+)
+
+func Test_getAuthProviders(t *testing.T) {
+
+	setupServer := func(response string) *httptest.Server {
+		return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
The servers created by this aren't `.Closed()` ?

Probably won't impact in the scope of tests.

Reading the code tho', it feels like you could create and either `defer .Close()` or `t.Cleanup()` and close?